### PR TITLE
Regenerate release notes for BC3

### PR DESCRIPTION
**Pull request by craigtaverner**
Regenerated release notes for BC3, and two new PRs were added. See generated docs at https://elasticsearch_87449.docs-preview.app.elstc.co/diff

### CODE REVIEW

**Review by lcawl — APPROVED** [review]
Thanks, docs LGTM

File: docs/reference/release-notes/8.3.0.asciidoc [hunk 1]
```diff
@@ -319,6 +320,7 @@ SQL::
 * Update dependency - JLine - to v 3.21.0 {es-pull}83767[#83767] (issue: {es-issue}83575[#83575])
 
 Search::
+* Update to public lucene 9.2.0 release {es-pull}87162[#87162]
```
Comment (pugnascotia) [inline docs/reference/release-notes/8.3.0.asciidoc:323]: @lcawl Should we remove the entries (and changelog files) for the Lucene snapshot upgrades? I would have thought that they're superseded by the proper 9.2.0 release?

Comment (craigtaverner) [reply docs/reference/release-notes/8.3.0.asciidoc:323]: Perhaps we should simply delete the changelog files for 86277 and 86852?
We could do this before the next build candidate and get the effects from regenerating the release notes, or we could do it as part of this PR here?

Comment (debadair) [reply docs/reference/release-notes/8.3.0.asciidoc:323]: Looking at the RNs for past releases, it looks like we removed the entries for the Snapshot upgrades, and just left the entry for the final upgrade. Removing the changelog files for the snapshot upgrades looks like the right thing to do IMO.

Comment (craigtaverner) [reply docs/reference/release-notes/8.3.0.asciidoc:323]: Great. I've removed the two changelog entries now.

**Review by pugnascotia — APPROVED** [review]

**Pull request merged**